server/container_create_linux.go

```diff
@@ -517,8 +517,11 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID, contai
 	if linux != nil {
 		resources := linux.GetResources()
 		if resources != nil {
-			specgen.SetLinuxResourcesCPUPeriod(uint64(resources.GetCpuPeriod()))
-			specgen.SetLinuxResourcesCPUQuota(resources.GetCpuQuota())
+			// only set period or quota if both are configured, or else the runtime will fail
+			if resources.GetCpuPeriod() != 0 && resources.GetCpuQuota() != 0 {
+				specgen.SetLinuxResourcesCPUPeriod(uint64(resources.GetCpuPeriod()))
+				specgen.SetLinuxResourcesCPUQuota(resources.GetCpuQuota())
+			}
```

Comment:
I believe instead of silently ignoring one setting if the other is not set, we should actually return an error here, as it's an invalid configuration (the two parameters depend on each other and it does not make sense to set one but not the other).